### ISSUE TITLE
Using the transform embedded in OpenIGTLink “IMAGE” messages 

### ISSUE THREAD

**Issue by Tiffas** [PlusToolkit/PlusLib#459]
Hi,

I am trying to use the transform embedded in OpenIGTLink “IMAGE” messages. In the Plus documentation about OpenIGTLink devices, it is specified about the **ImageMessageEmbeddedTransformName** attribute that :
_**ImageMessageEmbeddedTransformName** If IMAGE message is received and this attribute is defined then the transform embedded in the message will be recorded as a transform, with the specified name (e.g., “ImageToReference”)._

So I tried the following xml file:

<PlusConfiguration version="2.0">

	<DataCollection StartupDelaySec="1.0" >

		<DeviceSet 
			Name="PlusServer: Verasonics image sending"
			Description="Broadcasting acquired ultrasound video acquired on the Verasonics system through OpenIGTLink" />

			<Device
				Id="Verasonics"
				Type="OpenIGTLinkVideo"        
				MessageType="IMAGE"
				ServerAddress="172.31.1.2"
				ServerPort="18944"
				IgtlMessageCrcCheckEnabled="false"
				ReconnectOnReceiveTimeout="false"
				UseReceivedTimestamps="false"
				AcquisitionRate="40"
				ReceiveTimeoutSec="30"
				SendTimeoutSec="30"
				ImageMessageEmbeddedTransformName="ImageToReference">
				<DataSources>
					<DataSource Type="Video" Id="Video" PortUsImageOrientation="MF"  />
				</DataSources>
				<OutputChannels>
					<OutputChannel Id="VideoStream" VideoDataSourceId="Video" />
				</OutputChannels>
			</Device> 

	</DataCollection>

	<PlusOpenIGTLinkServer 
		MaxNumberOfIgtlMessagesToSend="1" 
		MaxTimeSpentWithProcessingMs="50" 
		ListeningPort="18944" 
		SendValidTransformsOnly="true" 
		OutputChannelId="VideoStream" > 
		<DefaultClientInfo> 
			<MessageTypes> 
				<Message Type="IMAGE" />
			</MessageTypes>
			<ImageNames>
				<Image Name="Image" EmbeddedTransformToFrame="Reference" />
			</ImageNames>
		</DefaultClientInfo>
	</PlusOpenIGTLinkServer>

</PlusConfiguration>

Here is the log of the errors I get with this file : 

[120418_161108_PlusLog.txt](https://github.com/PlusToolkit/PlusLib/files/2644378/120418_161108_PlusLog.txt)

I have no clue on what I did wrong.
Thanks for your help !

**Comment by Tiffas** [comment]
Same kind of lack of support of OpenIGTLink v1 and v2 as described in #504. In igsioTrackedFrame.cxx (version 2.7.0), in function member GetFrameTransformStatus(const igsioTransformName& frameTransformName, ToolStatus& status) if there is no transformStatus field the function return IGSIO_FAIL. @adamrankin I think that if a message has a header with V = 1, everything related to metadata and especially this transformStatus should be skipped (instead of treating the lack of this field as a special case). I think this would assure a better support of v1 and v2.

**Comment by adamrankin** [comment]
I agree. I don't have the bandwidth to do a comprehensive v1/v2 test, but I can fix bugs as you/others find them.
